lua/neo-tree/sources/filesystem/lib/filter_external.lua

```diff
@@ -43,133 +43,297 @@ local get_find_command = function(state)
   return state.find_command
 end
 
-M.find_files = function(opts)
-  local filters = opts.filtered_items
-  local limit = opts.limit or 200
-  local cmd = get_find_command(opts)
-  local path = opts.path
-  local full_path_words = opts.find_by_full_path_words
-  local regex, glob = opts.term, opts.term
-  local fuzzy_finder_mode = opts.fuzzy_finder_mode
+---@class FileTypes
+---@field file boolean
+---@field directory boolean
+---@field symlink boolean
+---@field socket boolean
+---@field pipe boolean
+---@field executable boolean
+---@field empty boolean
+---@field block boolean Only for `find`
+---@field character boolean Only for `find`
 
-  if full_path_words then
-    local words = utils.split(glob, " ")
-    regex = ".*" .. table.concat(words, ".*") .. ".*"
-  else
-    if glob ~= "*" then
-      if glob:sub(1) ~= "*" then
-        glob = "*" .. glob
-      end
-      if glob:sub(-1) ~= "*" then
-        glob = glob .. "*"
-      end
-    end
+---filter_files_external
+-- Spawns a filter command based on `cmd`
+---@param cmd string Command to execute. Use `get_find_command` most times.
+---@param path string Base directory to start the search.
+---@param glob string | nil If not nil, do glob search. Take precedence on `regex`
+---@param regex string | nil If not nil, do regex search if command supports. if glob ~= nil, ignored
+---@param full_path boolean If true, search agaist the absolute path
+---@param types FileTypes | nil Return only true filetypes. If nil, all are returned.
+---@param ignore { dotfiles: boolean?, gitignore: boolean? } If true, ignored from result. Default: false
+---@param limit? integer | nil Maximim number of results. nil will return everything.
+---@param find_args? string[] | table<string, string[]> Any additional options passed to command if any.
+---@param on_insert? fun(err: string, line: string): any Executed for each line of stdout and stderr.
+---@param on_exit? fun(return_val: table): any Executed at the end.
+M.filter_files_external = function(cmd, path, glob, regex, full_path, types, ignore, limit, find_args, on_insert, on_exit)
```

Comment:
```suggestion
M.filter_files_external = function(path, glob, regex, full_path, types, ignore, limit, find_args, on_insert, on_exit)
```
As described above, I think the `cmd` parameter should be removed.